ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/IgniteTestHelper.java

```diff
@@ -0,0 +1,131 @@
+import org.hibernate.ogm.model.spi.TupleSnapshot;
+import org.hibernate.ogm.persister.impl.OgmCollectionPersister;
+import org.hibernate.ogm.persister.impl.OgmEntityPersister;
+import org.hibernate.ogm.utils.TestableGridDialect;
+import org.hibernate.persister.collection.CollectionPersister;
+import org.hibernate.persister.entity.EntityPersister;
+
+/**
+ * @author Dmitriy Kozlov
+ */
+public class IgniteTestHelper implements TestableGridDialect {
+
+	@Override
+	public long getNumberOfEntities(SessionFactory sessionFactory) {
+		int entityCount = 0;
+		Set<IgniteCache<?, ?>> processedCaches = Collections.newSetFromMap( new IdentityHashMap<IgniteCache<?, ?>, Boolean>() );
+		for ( EntityPersister entityPersister : ( (SessionFactoryImplementor) sessionFactory ).getEntityPersisters().values() ) {
+			IgniteCache<?, ?> entityCache = getEntityCache( sessionFactory, ( (OgmEntityPersister) entityPersister ).getEntityKeyMetadata() );
+			if ( !processedCaches.contains( entityCache ) ) {
+				entityCount += entityCache.size(CachePeekMode.ALL);
+				processedCaches.add( entityCache );
+			}
+		}
+		return entityCount;
+	}
+	
+	@Override
+	public long getNumberOfAssociations(SessionFactory sessionFactory) {
+		int associationCount = 0;
+		Set<IgniteCache<?, ?>> processedCaches = Collections.newSetFromMap( new IdentityHashMap<IgniteCache<?, ?>, Boolean>() );
+		for ( CollectionPersister colleactionPersister : ( (SessionFactoryImplementor) sessionFactory ).getCollectionPersisters().values() ) {
+			IgniteCache<?, ?> associationCache = getAssociationCache( sessionFactory, ( (OgmCollectionPersister) colleactionPersister ).getAssociationKeyMetadata() );
+			if ( !processedCaches.contains( associationCache ) ) {
+				associationCount += associationCache.size(CachePeekMode.ALL);
+				processedCaches.add( associationCache );
+			}
+		}
+		return associationCount;
+	}
+
+	@Override
+	public long getNumberOfAssociations(SessionFactory sessionFactory, AssociationStorageType type) {
+		return 0;
+	}
+
+	@Override
+	public Map<String, Object> extractEntityTuple(SessionFactory sessionFactory, EntityKey key) {
+		IgniteCache<String, BinaryObject> cache = getEntityCache( sessionFactory, key.getMetadata() );
+		String cacheKey = getProvider( sessionFactory ).getKeyProvider().getEntityKeyString(key);
+
+		Map<String, Object> result = new HashMap<>();
+		Object po = cache.get(cacheKey);
+		
+		IgniteDialect igniteDialect = (IgniteDialect) ((SessionFactoryImplementor)sessionFactory).getServiceRegistry().getService(GridDialect.class);
+		TupleSnapshot snapshot = igniteDialect.createTupleSnapshot(po);
+		for (String fieldName : snapshot.getColumnNames()) {
+			result.put(fieldName, snapshot.get(fieldName));
+		}
+		
+		return result;
+	}
+
+	@Override
+	public boolean backendSupportsTransactions() {
+		return true;
+	}
+
+	@Override
+	public void dropSchemaAndDatabase(SessionFactory sessionFactory) {
+		// TODO че тут делать???
```

Comment:
This will be called by the test methods extending OgmTestCase, and it's executed before closing the session factory.
An in memory db I don't think has to do anything (when dealing with remote dbs we want to drop the db, for example)